src/Stack/Build/Source.hs

```diff
@@ -436,13 +436,13 @@ loadLocalPackage pp = do
 
 -- | Compare the current filesystem state to the cached information, and
 -- determine (1) if the files are dirty, and (2) the new cache values.
-checkBuildCache :: forall m. (MonadIO m)
+checkBuildCache :: HasEnvConfig env
                 => Map FilePath FileCacheInfo -- ^ old cache
                 -> [Path Abs File] -- ^ files in package
-                -> m (Set FilePath, Map FilePath FileCacheInfo)
+                -> RIO env (Set FilePath, Map FilePath FileCacheInfo)
 checkBuildCache oldCache files = do
   fileTimes <- fmap Map.fromList $ forM files $ \fp -> do
```

Comment:
Not a necessary part of your change, but I assume the name of `fileTimes` reflects some ancient history of this part of the code. Perhaps it should be refactored to be named `fileDigests` and this is an opportunity to do that?

Reply:
I intentionalyl wanted to keep the first draft very limited, I'm happy to clean this all up.

Reply:
I'll not get to a cleanup in this PR. Sorry.